### ISSUE TITLE
Title caser errors out

### ISSUE THREAD

**Issue by retorquere** [zotero/zotero#1617]
I have a user over at [BBT](https://github.com/retorquere/zotero-better-bibtex/issues/1107#issuecomment-450778959) where the citeproc title caser errors out. I have not been able to reproduce it locally, but I know where it happens; [tagDoppel.split](https://github.com/zotero/zotero/blob/master/chrome/content/zotero/xpcom/citeproc.js#L15185) for reasons I have not yet been able to figure out turns

```
Proc. 9th Int. Conf. Nuclear Engineering (ICONE-9)
```

into

```
{"tags":[".",".",".","(","-",")"],"strings":["Proc"," 9th Int"," Conf"," Nuclear Engineering ","ICONE","9",""],"origStrings":["Proc"," 9th Int"," Conf"," Nuclear Engineering ","ICONE","9",""]}`
```

for me, but into

```
{"tags":[".",".",".","(","-",")"],"strings":["Proc"," 9th Int"," Conf"," Nuclear Engineering ","ICONE","9",2.0809697299076e-311],"origStrings":["Proc"," 9th Int"," Conf"," Nuclear Engineering ","ICONE","9",2.0809697299076e-311]}
```

for @geekonloose

this issue popped up because BBT also uses the titlecaser, but it also shows up if @geekonloose just creates a Bibliograhy in `Chicago` style using Zotero without using BBT. I have stared at that bit of citeproc for hours but I honestly have not a single clue how this can happen.

**Comment by retorquere** [comment]
The regular error log says

```
[JavaScript Error: "TypeError: str.trim is not a function" {file: "chrome://zotero/content/xpcom/citeproc.js" line: 15375}]
CSL.Output.Formatters</title/config.capitaliseWords@chrome://zotero/content/xpcom/citeproc.js:15375:21
_textcaseEngine@chrome://zotero/content/xpcom/citeproc.js:15263:46
title@chrome://zotero/content/xpcom/citeproc.js:15407:16
CSL.Output.Queue.prototype.append@chrome://zotero/content/xpcom/citeproc.js:3820:26
CSL.Transform/getOutputFunction/<@chrome://zotero/content/xpcom/citeproc.js:12786:17
CSL.tokenExec@chrome://zotero/content/xpcom/citeproc.js:2032:21
CSL.expandMacro/func@chrome://zotero/content/xpcom/citeproc.js:2083:28
CSL.tokenExec@chrome://zotero/content/xpcom/citeproc.js:2032:21
CSL.getCite@chrome://zotero/content/xpcom/citeproc.js:5812:16
CSL.getBibliographyEntries@chrome://zotero/content/xpcom/citeproc.js:6126:38
CSL.Engine.prototype.makeBibliography@chrome://zotero/content/xpcom/citeproc.js:5941:11
makeFormattedBibliography@chrome://zotero/content/xpcom/cite.js:137:13
Zotero.Cite.makeFormattedBibliographyOrCitationList@chrome://zotero/content/xpcom/cite.js:79:23
Zotero_File_Interface</this.copyItemsToClipboard@chrome://zotero/content/fileInterface.js:653:17
_doBibliographyOptions@chrome://zotero/content/fileInterface.js:725:5
bibliographyFromItems@chrome://zotero/content/fileInterface.js:620:3
oncommand@chrome://zotero/content/standalone/standalone.xul:1:1


(4)(+0000000): SELECT IA.itemID FROM itemAttachments IA NATURAL JOIN items I LEFT JOIN itemData ID ON (IA.itemID=ID.itemID AND fieldID=1) LEFT JOIN itemDataValues IDV ON (ID.valueID=IDV.valueID) WHERE parentItemID=? AND linkMode NOT IN (?) AND IA.itemID NOT IN (SELECT itemID FROM deletedItems) ORDER BY contentType='application/pdf' DESC, value=? DESC, dateAdded ASC [259, 3, 'http://www.iaea.org/inis/collection/NCLCollectionStore/_Public/33/015/33015262.pdf']
(4)(+0000026): SELECT IA.itemID FROM itemAttachments IA NATURAL JOIN items I LEFT JOIN itemData ID ON (IA.itemID=ID.itemID AND fieldID=1) LEFT JOIN itemDataValues IDV ON (ID.valueID=IDV.valueID) WHERE parentItemID=? AND linkMode NOT IN (?) AND IA.itemID NOT IN (SELECT itemID FROM deletedItems) ORDER BY contentType='application/pdf' DESC, value=? DESC, dateAdded ASC [259, 3, 'http://www.iaea.org/inis/collection/NCLCollectionStore/_Public/33/015/33015262.pdf']
(4)(+0000014): SELECT IA.itemID FROM itemAttachments IA NATURAL JOIN items I LEFT JOIN itemData ID ON (IA.itemID=ID.itemID AND fieldID=1) LEFT JOIN itemDataValues IDV ON (ID.valueID=IDV.valueID) WHERE parentItemID=? AND linkMode NOT IN (?) AND IA.itemID NOT IN (SELECT itemID FROM deletedItems) ORDER BY contentType='application/pdf' DESC, value=? DESC, dateAdded ASC [259, 3, 'http://www.iaea.org/inis/collection/NCLCollectionStore/_Public/33/015/33015262.pdf']
(4)(+0000016): SELECT IA.itemID FROM itemAttachments IA NATURAL JOIN items I LEFT JOIN itemData ID ON (IA.itemID=ID.itemID AND fieldID=1) LEFT JOIN itemDataValues IDV ON (ID.valueID=IDV.valueID) WHERE parentItemID=? AND linkMode NOT IN (?) AND IA.itemID NOT IN (SELECT itemID FROM deletedItems) ORDER BY contentType='application/pdf' DESC, value=? DESC, dateAdded ASC [259, 3, 'http://www.iaea.org/inis/collection/NCLCollectionStore/_Public/33/015/33015262.pdf']
(3)(+0000020): MIME type application/pdf cannot be handled internally
(4)(+0000006): SELECT IA.itemID FROM itemAttachments IA NATURAL JOIN items I LEFT JOIN itemData ID ON (IA.itemID=ID.itemID AND fieldID=1) LEFT JOIN itemDataValues IDV ON (ID.valueID=IDV.valueID) WHERE parentItemID=? AND linkMode NOT IN (?) AND IA.itemID NOT IN (SELECT itemID FROM deletedItems) ORDER BY contentType='application/pdf' DESC, value=? DESC, dateAdded ASC [259, 3, 'http://www.iaea.org/inis/collection/NCLCollectionStore/_Public/33/015/33015262.pdf']
(4)(+0000050): SELECT IA.itemID FROM itemAttachments IA NATURAL JOIN items I LEFT JOIN itemData ID ON (IA.itemID=ID.itemID AND fieldID=1) LEFT JOIN itemDataValues IDV ON (ID.valueID=IDV.valueID) WHERE parentItemID=? AND linkMode NOT IN (?) AND IA.itemID NOT IN (SELECT itemID FROM deletedItems) ORDER BY contentType='application/pdf' DESC, value=? DESC, dateAdded ASC [259, 3, 'http://www.iaea.org/inis/collection/NCLCollectionStore/_Public/33/015/33015262.pdf']
(3)(+0007594): {better-bibtex +null} cache hit: {"itemID":259,"legacy":false,"skipChildItems":true}
(3)(+0000002): {better-bibtex +2} JournalAbbrev.get: getting from serialised
(3)(+0000000): {better-bibtex +0} JournalAbbrev.get: trying publicationTitle
(3)(+0000002): {better-bibtex +2} JournalAbbrev.get: trying reporter
(3)(+0000000): {better-bibtex +0} JournalAbbrev.get: trying code
(3)(+0000196): Alert:

An error occurred generating your bibliography. Please try again.
```

but I have an instrumented citeproc which gives more debug info.

**Comment by dstillman** [comment]
If you've narrowed this down to citeproc.js, you should report it to Frank in [that repo](https://github.com/Juris-M/citeproc-js).

But if it's something that can be reproduced in stock Zotero with BBT uninstalled, we'd also want a report in the [forums](https://forums.zotero.org) from @geekonloose with a <a href="https://www.zotero.org/support/reporting_problems">Report ID and steps to reproduce</a>.

(I have no idea how that would happen either, but we'd need exact steps to reproduce to investigate further.)

**Comment by geekonloose** [comment]
@dstillman Report ID is ```1510776339```.
Debug id is ```D1999891034```
Hope this helps to reproduce the issue.

**Comment by retorquere** [comment]
It can (and has been) be reproduced without any plugin installed, but only in this particular Zotero install. I have not been able to reproduce the error anywhere else.

**Comment by retorquere** [comment]
The instrumented citeproc in BBT was used to find the specific place where it happens, but for @geekonloose, the error reproduces the same without any plugins installed. I have not been able to reproduce it anywhere else, but for @geekonloose, it reproduces consistently (with no plugins installed)

**Comment by retorquere** [comment]
This is also the reason why I've reported it here - the error should simply not be possible (string.split just doesn't work that way) so it doesn't look to be a bug in citeproc but in some way a problem in the environment in which it is hosted. I've tested that bit of citeproc in Zotero, Firefox, Chrome and node and I can't reproduce it anywhere but @geekonloose's zotero install. But there it happens consistently.
